app/models/user.rb

```diff
@@ -5,14 +5,14 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
 
   has_one_attached :icon
-  validate :icon_image_type
+  validate :type
 
   private
 
-  def icon_image_type
+  def type
     return if icon.blob.nil?
 
-    format_allowed = %w[image/gif image/jpeg image/png]
-    errors.add :icon, :invalid_format, message: I18n.t('activerecord.models.errors.icon.invalid_format') unless format_allowed.include?(icon.blob.content_type)
+    allowed_types = %w[image/gif image/jpeg image/png]
+    errors.add :icon, :invalid_type, message: I18n.t('activerecord.models.errors.icon.invalid_type') unless allowed_types.include?(icon.blob.content_type)
```

Comment:
```suggestion
    errors.add :icon, :invalid_type unless allowed_types.include?(icon.blob.content_type)
```

のように書いた場合、Railsが自動的に命名規則に沿った翻訳を探しに行くので、それに合わせて翻訳キーを設定するといいかもしれません（てか、もしかすると上のコードのままでもうまくいくかも？）

あと、unlessがどこにかかるのかぱっと見わかりづらいので`()`を付けた方がベターかなと思いますー。

```suggestion
    errors.add(:icon, :invalid_type) unless allowed_types.include?(icon.blob.content_type)
```

Reply:
エラーの翻訳位置を命名規則に沿った形にして、tメソッドを省略しました。